### ISSUE TITLE
HEVC not working in Linux

### ISSUE THREAD

**Issue by octinhuh** [alvr-org/ALVR#706]
- [x] I carefully followed the instructions in the [README](https://github.com/alvr-org/ALVR/blob/master/README.md) and completed successfully the setup wizard.
- [x] I read the ALVR Wikis [here](https://github.com/polygraphene/ALVR/wiki) and [here](https://github.com/alvr-org/ALVR/wiki)

If your issue is about games not working, please open an issue for every game and list the game in the title.

### Bug reports:

Upon connection with the headset, the logs report `[INFO]	failed to create VAAPI encoder`, regardless of whether h264 or HEVC are selected in the settings. [log with hevc.txt](https://github.com/alvr-org/ALVR/files/6688679/log.with.hevc.txt)

The h264 setting works, albeit at around 10-20 fps. With HEVC enabled, a pixelated mess is displayed, shown below, at 0 fps. The server reports that it sends 5-10 fps, while the client is at 0.

I've tried changing mesa versions but ALVR doesn't seem to recognize my GPU's VAAPI encoder.

![unknown](https://user-images.githubusercontent.com/6424417/122807873-1b50aa80-d29a-11eb-877a-cd41b3761dbc.png)

#### Environment

Pop_OS 20.04 (also tried with Ubuntu 20.04 with the same results)

What hardware do you have:

- CPU: AMD Ryzen 5 1600
- GPU: AMD Radeon RX 580 (`OpenGL ES profile version string: OpenGL ES 3.2 Mesa 21.1.3 - kisak-mesa PPA`)

What version of ALVR and SteamVR have you tried. 
- ALVR version: [813b8b7](https://github.com/alvr-org/ALVR/commit/813b8b792289c093c5c3495ede9b1467e0e549c7)
- SteamVR Version 1.17.16 (1623188486)

**Comment by xytovl** [comment]
The artefact is what happens with hevc software encoding (not properly configured yet).

The RX580 is supposed to work with vaapi, what is the output of vainfo?

**Comment by octinhuh** [comment]
> The artefact is what happens with hevc software encoding (not properly configured yet).
> 
> The RX580 is supposed to work with vaapi, what is the output of vainfo?

```libva info: VA-API version 1.7.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_7
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.7 (libva 2.6.0)
vainfo: Driver version: Mesa Gallium driver 21.1.3 - kisak-mesa PPA for Radeon RX 580 Series (POLARIS10, DRM 3.40.0, 5.11.0-7614-generic, LLVM 12.0.0)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileNone                   :	VAEntrypointVideoProc```

It says I've got version 1.7.0 and libva 2.6.0, not too sure if this is what ALVR requires or not

**Comment by xytovl** [comment]
Your vaapi setup should work (VAEntrypointEncSlice is present both for hevc and h264).

There is certainly an issue during build, did you do the `cargo xtask build-server --release --bundle-ffmpeg` method? If so, do you have libva dev package?

The versions reported by vainfo are for the API itself (1.7), and the library that implements it (2.6.0). Library is not the newest one, but ffmpeg should work just fine with that one.

**Comment by octinhuh** [comment]
Ok I got it to work after recompiling. The `configure` script complained about not having `libdrm` at first which was really strange because `pkg-config` reported version 2.4.105 was installed. After probing some more with `--print-errors`, it said I didn't have `valgrind` installed (doh!). I've got a fresh install and that's something I didn't get around to. The reason ALVR had compiled before today without a working `libdrm` was because I must have done it before reinstalling my OS.

ALVR seems to be working now under HEVC for me. Not quite as good as on Windows, but time will tell
